File: Linear_Search.cpp
//Program to perform linear search on an array and print the index of first occurrence of the searched element

#include <iostream>
using namespace std;

int main()
{
   int n,s,i,A[1000];         //Assuming max size of array is 1000
   cout<<"Enter the no of elements";
   cin>>n;
   
   cout<<"Enter the elements";
   for(i=0;i<n;i++)
   {
       cin>>A[i];
       i++;
   }
   
   cout<<"Enter the element that you want to search";
   cin>>s;
   
   i=0;
   while(i<n)
   {
       if(A[i]==s)         //If current element of array is equal to the no that 
       break;             //we are searching, we can stop the searching process
       i++;
   }
   if(i<n)
   {
       cout<<"Element found at index"<<i;
   }
   else
   cout<<"Element is not present in the array";
   return 0;
}

File: Linear_Search.java
package Searching;

import java.util.*;

class Linear_Search
{
    int size;
    int a[];
    int search;
    
    public Linear_Search(int n)
    {
        size=n;
        a=new int[size];
    }
    public void input()
    {
        Scanner in=new Scanner(System.in);
        System.out.println("Enter the elements of the array:");
        for(int i=0;i<size;i++)
        a[i]=in.nextInt();
        
        System.out.println("Enter the search element:");
        search=in.nextInt();
    }
    public int search()
    {
        for(int i=0;i<size;i++)
        {
            if(a[i]==search)
            return i;
        }
        return -1;
    }
    public static void main(String args[])
    {
        Scanner in=new Scanner(System.in);
        
        System.out.println("Enter the size of the array:");
        int n=in.nextInt();
        
        Linear_Search ob=new Linear_Search(n);
        ob.input();
        int index=ob.search();
        System.out.println(index!=-1?"Search element found at "+index+" index":"Search element not found");
        
    }
}